router/pkg/config/config.schema.json

```diff
@@ -176,6 +172,30 @@
           "type": "boolean",
           "default": true,
           "description": "Forward the initial payload in the extensions payload when starting a subscription on a Subgraph. The default value is true."
+        },
+        "authentication": {
+          "type": "object",
+          "description": "The configuration used to authenticate the WebSocket connections.",
+          "additionalProperties": false,
+          "properties": {
+            "from_initial_payload": {
+              "type": "object",
+              "description": "The configuration used to tell the Router to look for the JWT (JSON Web Token) in the initial payload of the WebSocket Connection.",
+              "additionalProperties": false,
+              "properties": {
+                "secret_key": {
+                  "type": "string",
+                  "default": "Authorization",
+                  "description": "The name of the property holding the JWT (JSON Web Token) value. The default value is 'Authorization'"
+                },
+                "enabled": {
+                  "type": "boolean",
+                  "description": "This configuration indicates if the Router should look for the JWT (JSON Web Token) in the initial payload of the WebSocket Connection. The default value is 'false'.",
+                  "default": false
+                }
```

Comment:
We could add a property "export_token_to_client_request_header" with the following properties:
- enabled
- header_name (the header which we export the token into)

With this config, we could enable header forwarding in the engine, which would allow us to forward the token extracted from initial payload to sub-requests.

WDYT?

Reply:
sounds good 👍

Reply:
It seems like "header_name" is a reserved word or something because it did not let me use it, so I used "header_key" instead.